dbms/src/IO/Compression/CompressionCodecFactory.cpp

```diff
@@ -0,0 +1,214 @@
+{
+    // ZSTD level is in the range [1, 22], the maximum size of the map is 22.
+    static std::vector<CompressionCodecPtr> zstd_map = {
+        std::make_shared<CompressionCodecZSTD>(1),  std::make_shared<CompressionCodecZSTD>(2),
+        std::make_shared<CompressionCodecZSTD>(3),  std::make_shared<CompressionCodecZSTD>(4),
+        std::make_shared<CompressionCodecZSTD>(5),  std::make_shared<CompressionCodecZSTD>(6),
+        std::make_shared<CompressionCodecZSTD>(7),  std::make_shared<CompressionCodecZSTD>(8),
+        std::make_shared<CompressionCodecZSTD>(9),  std::make_shared<CompressionCodecZSTD>(10),
+        std::make_shared<CompressionCodecZSTD>(11), std::make_shared<CompressionCodecZSTD>(12),
+        std::make_shared<CompressionCodecZSTD>(13), std::make_shared<CompressionCodecZSTD>(14),
+        std::make_shared<CompressionCodecZSTD>(15), std::make_shared<CompressionCodecZSTD>(16),
+        std::make_shared<CompressionCodecZSTD>(17), std::make_shared<CompressionCodecZSTD>(18),
+        std::make_shared<CompressionCodecZSTD>(19), std::make_shared<CompressionCodecZSTD>(20),
+        std::make_shared<CompressionCodecZSTD>(21), std::make_shared<CompressionCodecZSTD>(22),
+    };
+    return zstd_map[setting.level - 1];
+}
+
+template <>
+CompressionCodecPtr CompressionCodecFactory::getStaticCodec<CompressionCodecNone>(const CompressionSetting &)
+{
+    static auto none = std::make_shared<CompressionCodecNone>();
+    return none;
+}
+
+#if USE_QPL
+template <>
+CompressionCodecPtr CompressionCodecFactory::getStaticCodec<CompressionCodecDeflateQpl>(const CompressionSetting &)
+{
+    static auto qpl = std::make_shared<CompressionCodecDeflateQpl>();
+    return qpl;
+}
+#endif
+
+
+template <bool IS_DECOMPRESS>
+CompressionCodecPtr CompressionCodecFactory::create(const CompressionSetting & setting)
+{
+    // LZ4 and LZ4HC have the same format, the difference is only in compression.
+    // So they have the same method byte.
+    if (setting.method == CompressionMethod::LZ4HC)
+        return getStaticCodec<CompressionCodecLZ4HC>(setting);
+    if (setting.method_byte == CompressionMethodByte::LZ4)
+        return getStaticCodec<CompressionCodecLZ4>(setting);
+    if (setting.method_byte == CompressionMethodByte::ZSTD)
+        return getStaticCodec<CompressionCodecZSTD>(setting);
+    if (setting.method_byte == CompressionMethodByte::NONE)
+        return getStaticCodec<CompressionCodecNone>(setting);
+
+#if USE_QPL
+    if (setting.method_byte == CompressionMethodByte::QPL)
+        return getStaticCodec<CompressionCodecDeflateQpl>(setting);
+#endif
+
+    if constexpr (!IS_DECOMPRESS)
+    {
+        // If method_byte is Lightweight, use LZ4 codec for non-integral types
+        // If method_byte is DeltaFOR/RunLength/FOR, since we do not support use these methods independently, so there must be another codec, use it directly.
```

Comment:
```suggestion
        // If method_byte is Lightweight, use LZ4 codec for non-integral types
        // If method_byte is DeltaFOR/RunLength/FOR, since we do not support use these methods independently,
        // there must be another codec to compress data. Use that compress codec directly.
```